app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala

```diff
@@ -120,12 +124,11 @@ class IncomeUpdateCalculatorController @Inject()(
       journeyCacheService
         .mandatoryJourneyValues(
           UpdateIncome_NameKey,
-          UpdateIncome_IdKey,
           s"$UpdateIncome_ConfirmedNewAmountKey-$empId",
           UpdateIncome_IncomeTypeKey)
         .getOrFail
-        .flatMap { mandatoryJourneyValues =>
-          val incomeName :: incomeId :: newAmount :: incomeType :: Nil = mandatoryJourneyValues.toList
+        .map { mandatoryJourneyValues =>
+          val incomeName :: newAmount :: incomeType = mandatoryJourneyValues.toList
```

Comment:
if u remove that Nil then incomType is a List which means the if below will never satisfy the condition: 
`val vm = if (incomeType == TaiConstants.IncomeTypePension) {`